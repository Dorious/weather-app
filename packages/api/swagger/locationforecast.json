{
  "basePath": "/weatherapi/locationforecast/2.0",
  "host": "api.met.no",
  "schemes": ["https"],
  "definitions": {
    "ForecastTimeStep": {
      "type": "object",
      "required": [
        "time",
        "data"
      ],
      "properties": {
        "time": {
          "description": "The time these forecast values are valid for. Timestamp in format YYYY-MM-DDThh:mm:ssZ (ISO 8601)",
          "example": "2019-12-03T14:00:00Z",
          "type": "string"
        },
        "data": {
          "type": "object",
          "required": [
            "instant"
          ],
          "properties": {
            "instant": {
              "properties": {
                "details": {
                  "$ref": "#/definitions/ForecastTimeInstant"
                }
              },
              "description": "Parameters which applies to this exact point in time",
              "type": "object"
            },
            "next_12_hours": {
              "description": "Parameters with validity times over twelve hours. Will not exist for all time steps.",
              "properties": {
                "details": {
                  "$ref": "#/definitions/ForecastTimePeriod"
                },
                "summary": {
                  "$ref": "#/definitions/ForecastSummary"
                }
              },
              "required": [
                "summary",
                "details"
              ],
              "type": "object"
            },
            "next_1_hours": {
              "required": [
                "summary",
                "details"
              ],
              "type": "object",
              "description": "Parameters with validity times over one hour. Will not exist for all time steps.",
              "properties": {
                "details": {
                  "$ref": "#/definitions/ForecastTimePeriod"
                },
                "summary": {
                  "$ref": "#/definitions/ForecastSummary"
                }
              }
            },
            "next_6_hours": {
              "type": "object",
              "required": [
                "summary",
                "details"
              ],
              "properties": {
                "summary": {
                  "$ref": "#/definitions/ForecastSummary"
                },
                "details": {
                  "$ref": "#/definitions/ForecastTimePeriod"
                }
              },
              "description": "Parameters with validity times over six hours. Will not exist for all time steps."
            }
          },
          "description": "Forecast for a specific time"
        }
      },
      "title": "Forecast for a specific time step"
    },
    "METJSONForecast": {
      "properties": {
        "type": {
          "enum": [
            "Feature"
          ],
          "example": "Feature",
          "type": "string"
        },
        "geometry": {
          "$ref": "#/definitions/PointGeometry"
        },
        "properties": {
          "$ref": "#/definitions/Forecast"
        }
      },
      "title": "GeoJSON Forecast Timeseries",
      "type": "object",
      "required": [
        "type",
        "geometry",
        "properties"
      ]
    },
    "ForecastUnits": {
      "type": "object",
      "properties": {
        "air_temperature": {
          "type": "string",
          "example": "C"
        },
        "probability_of_thunder": {
          "example": "%",
          "type": "string"
        },
        "dew_point_temperature": {
          "example": "C",
          "type": "string"
        },
        "wind_from_direction": {
          "type": "string",
          "example": "degrees"
        },
        "wind_speed": {
          "example": "m/s",
          "type": "string"
        },
        "precipitation_amount": {
          "type": "string",
          "example": "mm"
        },
        "cloud_area_fraction_high": {
          "type": "string",
          "example": "%"
        },
        "precipitation_amount_max": {
          "example": "mm",
          "type": "string"
        },
        "air_temperature_max": {
          "example": "C",
          "type": "string"
        },
        "probability_of_precipitation": {
          "example": "%",
          "type": "string"
        },
        "cloud_area_fraction": {
          "example": "%",
          "type": "string"
        },
        "air_pressure_at_sea_level": {
          "type": "string",
          "example": "hPa"
        },
        "cloud_area_fraction_low": {
          "example": "%",
          "type": "string"
        },
        "cloud_area_fraction_medium": {
          "type": "string",
          "example": "%"
        },
        "ultraviolet_index_clear_sky_max": {
          "type": "string",
          "example": "1"
        },
        "fog_area_fraction": {
          "type": "string",
          "example": "%"
        },
        "precipitation_amount_min": {
          "example": "mm",
          "type": "string"
        },
        "wind_speed_of_gust": {
          "example": "m/s",
          "type": "string"
        },
        "air_temperature_min": {
          "type": "string",
          "example": "C"
        },
        "relative_humidity": {
          "example": "%",
          "type": "string"
        }
      }
    },
    "Forecast": {
      "title": "Forecast timeseries",
      "properties": {
        "meta": {
          "properties": {
            "units": {
              "$ref": "#/definitions/ForecastUnits"
            },
            "updated_at": {
              "description": "Update time for this forecast",
              "type": "string",
              "example": "2019-12-03T13:52:13Z"
            }
          },
          "required": [
            "updated_at",
            "units"
          ],
          "type": "object"
        },
        "timeseries": {
          "type": "array",
          "items": {
            "$ref": "#/definitions/ForecastTimeStep"
          }
        }
      },
      "required": [
        "meta",
        "timeseries"
      ],
      "type": "object"
    },
    "ForecastTimeInstant": {
      "description": "Weather parameters valid for a specific point in time.",
      "properties": {
        "wind_speed": {
          "example": 5.9,
          "type": "number",
          "description": "Speed of wind"
        },
        "cloud_area_fraction_medium": {
          "example": 95.2,
          "type": "number",
          "description": "Amount of sky covered by clouds at medium elevation."
        },
        "dew_point_temperature": {
          "description": "Dew point temperature at sea level",
          "type": "number",
          "example": 8.1
        },
        "wind_from_direction": {
          "type": "number",
          "example": 121.3,
          "description": "The directon which moves towards"
        },
        "air_pressure_at_sea_level": {
          "description": "Air pressure at sea level",
          "type": "number",
          "example": 1017.23
        },
        "cloud_area_fraction_low": {
          "description": "Amount of sky covered by clouds at low elevation.",
          "example": 95.2,
          "type": "number"
        },
        "air_temperature": {
          "example": 17.1,
          "type": "number",
          "description": "Air temperature"
        },
        "cloud_area_fraction": {
          "description": "Amount of sky covered by clouds.",
          "example": 95.2,
          "type": "number"
        },
        "relative_humidity": {
          "description": "Amount of humidity in the air.",
          "example": 81.1,
          "type": "number"
        },
        "wind_speed_of_gust": {
          "example": 15.9,
          "type": "number",
          "description": "Speed of wind gust"
        },
        "fog_area_fraction": {
          "example": 95.2,
          "type": "number",
          "description": "Amount of area covered by fog."
        },
        "cloud_area_fraction_high": {
          "type": "number",
          "example": 95.2,
          "description": "Amount of sky covered by clouds at high elevation."
        }
      },
      "type": "object"
    },
    "ForecastSummary": {
      "description": "Summary of weather conditions.",
      "properties": {
        "symbol_code": {
          "$ref": "#/definitions/WeatherSymbol"
        }
      },
      "required": [
        "symbol_code"
      ],
      "type": "object"
    },
    "PointGeometry": {
      "properties": {
        "type": {
          "type": "string",
          "enum": [
            "Point"
          ]
        },
        "coordinates": {
          "description": "[longitude, latitude, altitude]. All numbers in decimal.",
          "items": {
            "type": "number"
          },
          "minItems": 2,
          "type": "array",
          "example": [
            60.5,
            11.59,
            1001
          ]
        }
      },
      "title": "GeoJSON point type",
      "type": "object",
      "required": [
        "type",
        "coordinates"
      ]
    },
    "WeatherSymbol": {
      "example": "clearsky_day",
      "type": "string",
      "enum": [
        "clearsky_day",
        "clearsky_night",
        "clearsky_polartwilight",
        "fair_day",
        "fair_night",
        "fair_polartwilight",
        "lightssnowshowersandthunder_day",
        "lightssnowshowersandthunder_night",
        "lightssnowshowersandthunder_polartwilight",
        "lightsnowshowers_day",
        "lightsnowshowers_night",
        "lightsnowshowers_polartwilight",
        "heavyrainandthunder",
        "heavysnowandthunder",
        "rainandthunder",
        "heavysleetshowersandthunder_day",
        "heavysleetshowersandthunder_night",
        "heavysleetshowersandthunder_polartwilight",
        "heavysnow",
        "heavyrainshowers_day",
        "heavyrainshowers_night",
        "heavyrainshowers_polartwilight",
        "lightsleet",
        "heavyrain",
        "lightrainshowers_day",
        "lightrainshowers_night",
        "lightrainshowers_polartwilight",
        "heavysleetshowers_day",
        "heavysleetshowers_night",
        "heavysleetshowers_polartwilight",
        "lightsleetshowers_day",
        "lightsleetshowers_night",
        "lightsleetshowers_polartwilight",
        "snow",
        "heavyrainshowersandthunder_day",
        "heavyrainshowersandthunder_night",
        "heavyrainshowersandthunder_polartwilight",
        "snowshowers_day",
        "snowshowers_night",
        "snowshowers_polartwilight",
        "fog",
        "snowshowersandthunder_day",
        "snowshowersandthunder_night",
        "snowshowersandthunder_polartwilight",
        "lightsnowandthunder",
        "heavysleetandthunder",
        "lightrain",
        "rainshowersandthunder_day",
        "rainshowersandthunder_night",
        "rainshowersandthunder_polartwilight",
        "rain",
        "lightsnow",
        "lightrainshowersandthunder_day",
        "lightrainshowersandthunder_night",
        "lightrainshowersandthunder_polartwilight",
        "heavysleet",
        "sleetandthunder",
        "lightrainandthunder",
        "sleet",
        "lightssleetshowersandthunder_day",
        "lightssleetshowersandthunder_night",
        "lightssleetshowersandthunder_polartwilight",
        "lightsleetandthunder",
        "partlycloudy_day",
        "partlycloudy_night",
        "partlycloudy_polartwilight",
        "sleetshowersandthunder_day",
        "sleetshowersandthunder_night",
        "sleetshowersandthunder_polartwilight",
        "rainshowers_day",
        "rainshowers_night",
        "rainshowers_polartwilight",
        "snowandthunder",
        "sleetshowers_day",
        "sleetshowers_night",
        "sleetshowers_polartwilight",
        "cloudy",
        "heavysnowshowersandthunder_day",
        "heavysnowshowersandthunder_night",
        "heavysnowshowersandthunder_polartwilight",
        "heavysnowshowers_day",
        "heavysnowshowers_night",
        "heavysnowshowers_polartwilight"
      ],
      "description": "A identifier that sums up the weather condition for this time period. May be used with https://api.met.no/weatherapi/weathericon/2.0/."
    },
    "ForecastTimePeriod": {
      "type": "object",
      "properties": {
        "ultraviolet_index_clear_sky_max": {
          "example": 1,
          "type": "number",
          "description": "Maximum ultraviolet index if sky is clear"
        },
        "precipitation_amount": {
          "description": "Best estimate for amount of precipitation for this period",
          "type": "number",
          "example": 1.71
        },
        "precipitation_amount_max": {
          "type": "number",
          "example": 4.32,
          "description": "Maximum amount of precipitation for this period"
        },
        "precipitation_amount_min": {
          "example": 4.32,
          "type": "number",
          "description": "Minimum amount of precipitation for this period"
        },
        "air_temperature_min": {
          "description": "Minimum air temperature in period",
          "type": "number",
          "example": 11.1
        },
        "probability_of_precipitation": {
          "type": "number",
          "example": 37,
          "description": "Probability of any precipitation coming for this period"
        },
        "air_temperature_max": {
          "description": "Maximum air temperature in period",
          "type": "number",
          "example": 17.1
        },
        "probability_of_thunder": {
          "type": "number",
          "example": 54.32,
          "description": "Probability of any thunder coming for this period"
        }
      },
      "description": "Weather parameters valid for a specified time period."
    }
  },
  "swagger": "2.0",
  "paths": {
    "/compact.{format}": {
      "get": {
        "parameters": [
          {
            "description": "Whole meters above sea level",
            "name": "altitude",
            "in": "query",
            "format": "integer",
            "required": false,
            "type": "integer"
          },
          {
            "type": "number",
            "required": true,
            "format": "float",
            "in": "query",
            "description": "Latitude",
            "name": "lat"
          },
          {
            "description": "Longitude",
            "name": "lon",
            "in": "query",
            "format": "float",
            "required": true,
            "type": "number"
          },
          {
            "format": "string",
            "in": "path",
            "description": "format code (file extension)",
            "name": "format",
            "enum": [
              "json"
            ],
            "type": "string",
            "required": true
          }
        ],
        "description": "Weather forecast for a specified place",
        "produces": [
          "application/json;charset=UTF-8"
        ],
        "x-mojo-name": "run",
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "$ref": "#/definitions/METJSONForecast"
            }
          }
        },
        "tags": [
          "data"
        ]
      }
    },
    "/classic": {
      "get": {
        "description": "Weather forecast for a specified place",
        "produces": [
          "application/xml"
        ],
        "parameters": [
          {
            "in": "query",
            "description": "Whole meters above sea level",
            "name": "altitude",
            "format": "integer",
            "required": false,
            "type": "integer"
          },
          {
            "required": true,
            "type": "number",
            "name": "lat",
            "description": "Latitude",
            "in": "query",
            "format": "float"
          },
          {
            "name": "lon",
            "description": "Longitude",
            "in": "query",
            "format": "float",
            "required": true,
            "type": "number"
          }
        ],
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "type": "string"
            }
          }
        },
        "tags": [
          "data"
        ],
        "x-mojo-name": "run"
      }
    },
    "/classic.{format}": {
      "get": {
        "parameters": [
          {
            "required": false,
            "type": "integer",
            "description": "Whole meters above sea level",
            "in": "query",
            "name": "altitude",
            "format": "integer"
          },
          {
            "description": "Latitude",
            "in": "query",
            "name": "lat",
            "format": "float",
            "required": true,
            "type": "number"
          },
          {
            "in": "query",
            "description": "Longitude",
            "name": "lon",
            "format": "float",
            "required": true,
            "type": "number"
          },
          {
            "format": "string",
            "description": "format code (file extension)",
            "name": "format",
            "in": "path",
            "enum": [
              "xml"
            ],
            "type": "string",
            "required": true
          }
        ],
        "produces": [
          "application/xml"
        ],
        "description": "Weather forecast for a specified place",
        "x-mojo-name": "run",
        "tags": [
          "data"
        ],
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "type": "string"
            }
          }
        }
      }
    },
    "/schema": {
      "get": {
        "description": "Schema for XML data",
        "produces": [
          "text/xml"
        ],
        "x-mojo-name": "schema",
        "responses": {
          "200": {
            "description": "OK"
          }
        },
        "tags": [
          "metadata"
        ]
      }
    },
    "/complete": {
      "get": {
        "tags": [
          "data"
        ],
        "responses": {
          "200": {
            "schema": {
              "$ref": "#/definitions/METJSONForecast"
            },
            "description": "Success"
          }
        },
        "x-mojo-name": "run",
        "produces": [
          "application/json;charset=UTF-8"
        ],
        "description": "Weather forecast for a specified place",
        "parameters": [
          {
            "description": "Whole meters above sea level",
            "in": "query",
            "name": "altitude",
            "format": "integer",
            "required": false,
            "type": "integer"
          },
          {
            "type": "number",
            "required": true,
            "format": "float",
            "name": "lat",
            "description": "Latitude",
            "in": "query"
          },
          {
            "required": true,
            "type": "number",
            "description": "Longitude",
            "in": "query",
            "name": "lon",
            "format": "float"
          }
        ]
      }
    },
    "/status": {
      "get": {
        "parameters": [],
        "produces": [
          "application/json;charset=UTF-8"
        ],
        "description": "Weather forecast for a specified place",
        "x-mojo-name": "run",
        "tags": [
          "data"
        ],
        "responses": {
          "200": {
            "schema": {
              "type": "string"
            },
            "description": "Success"
          }
        }
      }
    },
    "/status.{format}": {
      "get": {
        "parameters": [
          {
            "type": "string",
            "enum": [
              "json"
            ],
            "required": true,
            "format": "string",
            "name": "format",
            "description": "format code (file extension)",
            "in": "path"
          }
        ],
        "produces": [
          "application/json;charset=UTF-8"
        ],
        "description": "Weather forecast for a specified place",
        "x-mojo-name": "run",
        "tags": [
          "data"
        ],
        "responses": {
          "200": {
            "schema": {
              "type": "string"
            },
            "description": "Success"
          }
        }
      }
    },
    "/compact": {
      "get": {
        "x-mojo-name": "run",
        "tags": [
          "data"
        ],
        "responses": {
          "200": {
            "description": "Success",
            "schema": {
              "$ref": "#/definitions/METJSONForecast"
            }
          }
        },
        "parameters": [
          {
            "description": "Whole meters above sea level",
            "name": "altitude",
            "in": "query",
            "format": "integer",
            "required": false,
            "type": "integer"
          },
          {
            "required": true,
            "type": "number",
            "description": "Latitude",
            "name": "lat",
            "in": "query",
            "format": "float"
          },
          {
            "type": "number",
            "required": true,
            "format": "float",
            "name": "lon",
            "description": "Longitude",
            "in": "query"
          }
        ],
        "produces": [
          "application/json;charset=UTF-8"
        ],
        "description": "Weather forecast for a specified place"
      }
    },
    "/complete.{format}": {
      "get": {
        "produces": [
          "application/json;charset=UTF-8"
        ],
        "description": "Weather forecast for a specified place",
        "parameters": [
          {
            "format": "integer",
            "description": "Whole meters above sea level",
            "in": "query",
            "name": "altitude",
            "type": "integer",
            "required": false
          },
          {
            "required": true,
            "type": "number",
            "in": "query",
            "description": "Latitude",
            "name": "lat",
            "format": "float"
          },
          {
            "description": "Longitude",
            "in": "query",
            "name": "lon",
            "format": "float",
            "required": true,
            "type": "number"
          },
          {
            "enum": [
              "json"
            ],
            "type": "string",
            "required": true,
            "format": "string",
            "name": "format",
            "description": "format code (file extension)",
            "in": "path"
          }
        ],
        "tags": [
          "data"
        ],
        "responses": {
          "200": {
            "schema": {
              "$ref": "#/definitions/METJSONForecast"
            },
            "description": "Success"
          }
        },
        "x-mojo-name": "run"
      }
    },
    "/healthz": {
      "get": {
        "responses": {
          "200": {
            "description": "OK"
          }
        },
        "tags": [
          "metadata"
        ],
        "x-mojo-name": "healthz",
        "description": "Check health status for product",
        "produces": [
          "text/plain"
        ]
      }
    }
  },
  "info": {
    "contact": {
      "url": "http://met.no",
      "name": "Developer Support",
      "email": "weatherapi-adm@met.no"
    },
    "termsOfService": "/doc/TermsOfService",
    "license": {
      "name": "CC BY 4.0",
      "url": "http://creativecommons.org/licenses/by/4.0"
    },
    "title": "Locationforecast",
    "description": "Weather forecast for a specified place",
    "version": "2.0"
  }
}