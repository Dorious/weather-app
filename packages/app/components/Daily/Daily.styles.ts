import styled from 'styled-components';

export const Container = styled.section`
  border-top: 1px solid #fff;
  padding: 1rem 0;
`;

export const Day = styled.div`
  max-width: 600px;
  padding: 1rem 2rem;
  font-size: 1.2rem;
  margin: 0 auto;
  align-items: center;
  display: flex;
  border-bottom: 1px solid rgba(255, 255, 255, 0.2);

  :last-child {
    border-bottom: 0;
  }
`;

export const DayName = styled.h3`
  margin: 0;
  padding: 0;
  font-weight: normal;
  white-space: nowrap;
  min-width: 10rem;
`;

export const Symbol = styled.div`
  margin-right: 3rem;
  flex-grow: 1;

  > img {
    width: 3rem;
    height: 3rem;
  }
`;

export const Temperatures = styled.div`
  white-space: nowrap;
  * {
    width: 2rem;
    display: inline-block;
  }

  b {
    font-weight: normal;
  }
  span {
    margin-left: 1rem;
    opacity: 0.6;
  }
`;
