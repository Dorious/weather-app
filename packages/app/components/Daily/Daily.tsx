import { memo, useMemo } from 'react';
import { ForecastTimeStep } from '@dorious/weather-api/dist/locationforecast';

import * as S from './Daily.styles';
import Symbol from '../Symbol';

export type DailyProps = {
  timeseries: ForecastTimeStep[];
};

export default memo(function Daily({ timeseries }: DailyProps): JSX.Element {
  const dailyTimeseries = useMemo(() => {
    if (!timeseries) return null;

    const seriesByDate: Record<number, ForecastTimeStep> = {};

    timeseries.forEach((series) => {
      const date = new Date(new Date(series.time).toDateString()).getTime();

      if (!seriesByDate[date]) {
        seriesByDate[date] = series;
      } else {
        if (!seriesByDate[date].data.next_1_hours)
          seriesByDate[date].data.next_1_hours = series.data.next_1_hours;
        if (!seriesByDate[date].data.next_6_hours)
          seriesByDate[date].data.next_6_hours = series.data.next_6_hours;
      }
    });

    return seriesByDate;
  }, [timeseries]);

  if (!dailyTimeseries) return null;

  return (
    <S.Container>
      {Object.values(dailyTimeseries).map((series) => {
        const { air_temperature_max, air_temperature_min } =
          series.data.next_6_hours?.details || {};
        const { symbol_code } =
          series.data.next_1_hours?.summary ||
          series.data.next_6_hours?.summary ||
          {};

        return (
          <S.Day key={series.time}>
            <S.DayName>
              {new Date(series.time).toLocaleString('en-us', {
                weekday: 'long',
              })}
            </S.DayName>
            <S.Symbol>
              <Symbol code={symbol_code} />
            </S.Symbol>
            <S.Temperatures>
              <b>{Math.round(air_temperature_max)}&deg;</b>{' '}
              <span>{Math.round(air_temperature_min)}&deg;</span>
            </S.Temperatures>
          </S.Day>
        );
      })}
    </S.Container>
  );
});
