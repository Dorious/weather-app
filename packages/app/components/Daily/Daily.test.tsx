import { render } from '@testing-library/react';
import Daily from '.';

import { ForecastTimeStep } from '@dorious/weather-api/dist/locationforecast';

import weatherData from '../../__mocks__/weatherData.json';

describe('Hourly', () => {
  it('should render properly', () => {
    const { container } = render(
      <Daily
        timeseries={
          weatherData.data.properties.timeseries as ForecastTimeStep[]
        }
      />
    );
    expect(container).toMatchSnapshot();
  });
});
