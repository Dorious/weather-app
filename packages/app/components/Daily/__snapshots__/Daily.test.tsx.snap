// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hourly should render properly 1`] = `
<div>
  <section
    class="Dailystyles__Container-sc-1i15yny-0 fatvBA"
  >
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Monday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Cloudy"
          src="/symbols/svg/cloudy.svg"
          title="Cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          7
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Tuesday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          4
          °
        </b>
         
        <span>
          2
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Wednesday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          3
          °
        </b>
         
        <span>
          2
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Thursday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Cloudy"
          src="/symbols/svg/cloudy.svg"
          title="Cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          3
          °
        </b>
         
        <span>
          2
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Friday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          5
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Saturday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          6
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Sunday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          6
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Monday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          6
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Tuesday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Fair"
          src="/symbols/svg/fair_night.svg"
          title="Fair"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          7
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
    <div
      class="Dailystyles__Day-sc-1i15yny-1 bEMxJL"
    >
      <h3
        class="Dailystyles__DayName-sc-1i15yny-2 gmDgIC"
      >
        Wednesday
      </h3>
      <div
        class="Dailystyles__Symbol-sc-1i15yny-3 dkvAxZ"
      >
        <img
          alt="Partly cloudy"
          src="/symbols/svg/partlycloudy_night.svg"
          title="Partly cloudy"
        />
      </div>
      <div
        class="Dailystyles__Temperatures-sc-1i15yny-4 dlTnNH"
      >
        <b>
          6
          °
        </b>
         
        <span>
          4
          °
        </span>
      </div>
    </div>
  </section>
</div>
`;
